builder/parser/line_parsers.go

```diff
@@ -195,6 +192,44 @@ func parseLabel(rest string) (*Node, map[string]bool, error) {
 	return parseNameVal(rest, "LABEL")
 }
 
+// parses a statement with multiple keywords definition and name-value
+// assignment, like `name1= name2=value`.
+// Note that this is a stricter format than the old format of assignment, in a way
+// that it only allows assignment of the form `name=value`. And that a name
+// definition or assignment to an empty value is designated by `name=`.
+func parseNameOrNameVal(rest string) (*Node, map[string]bool, error) {
+	words := parseWords(rest)
+	if len(words) == 0 {
+		return nil, nil, nil
+	}
+
+	var (
+		rootnode *Node
+		prevNode *Node
+	)
+	for i, word := range words {
+		if !strings.Contains(word, "=") {
+			return nil, nil, fmt.Errorf("Syntax error - can't find = in %q. Must be of the form: name=value", word)
```

Comment:
sure, I can move the logic to the `arg()`, but will wait on what @tiborvass has to say regarding the comment below (https://github.com/docker/docker/pull/15182#discussion_r39046060).

I think based on what we decide there, this logic will change again.